docs/contracts.rst

```diff
@@ -584,6 +591,89 @@ Please ensure you test your fallback function thoroughly to ensure the execution
         }
     }
 
+.. index:: ! overload
+
+.. _overload-function:
+
+Function Overloading
+====================
+A Contract can have multiple functions of the same name but with different arguments.
+This also applies to inherited functions. The following example shows overloading of the
+``Z`` function in the scope of contract ``A``.
+
+::
+
+    pragma solidity ^0.4.0;
+
+    contract A {
+        function z(uint _in) public pure returns (uint out)
+        {
+            out = 1;
+        }
+
+        function z(uint _in, bytes32 _key) public pure returns (uint out)
+        {
+            out = 2;
+        }
+    }
+
+Overloaded functions are also present in the external interface. It is an error if two
+externally visible functions differ by their Solidity types but not by their external types.
+
+::
+
+    pragma solidity ^0.4.0;
+
+    contract A {
+        function z(B _in) public pure returns (B out)
+        {
+            out = _in;
+        }
+
+        function z(address _in) public pure returns (address out)
```

Comment:
Correctly detected by the compiler:
```
/tmp/tmp.If4S5vMgWU/test_9299c7c399da4cec7ae3fe218d6b8fbe4474ab118851fcf2eeeb49c8ecf50eb5.sol:8:9: Error: Function overload clash during conversion to external types for arguments.
        function z(address _in) public pure returns (address out)
        ^
```

This example will need the comment right above the pragma: `// This will not compile`